sphinxcontrib/needs/utils.py

```diff
@@ -145,6 +156,22 @@ def rstjinja(app: Sphinx, docname: str, source):
     source[0] = rendered
 
 
+def import_prefix_link_edit(env, needs, id_prefix):
+    needs_ids = needs.keys()
+
+    for key, need in needs.items():
+        for id in needs_ids:
+            # Manipulate links in all link types
+            for extra_link in env.config.needs_extra_links:
+                if extra_link["option"] in need.keys() and id in need[extra_link["option"]]:
+                    for n, link in enumerate(need[extra_link["option"]]):
+                        if id == link:
+                            need[extra_link["option"]][n] = "".join([id_prefix, id])
```

Comment:
i think this might be better done with a list comprehension. This is a very confusing bit of logic

Reply:
Maybe, but it works and is tested (copy & past from needimport code).
So I don't touch it with this PR.

Reply:
if it's copy-pasted, can it be extracted into a function (to avoid duplicating logic!)?